src/riak_kv_sup.erl

```diff
@@ -98,10 +98,14 @@ init([]) ->
                 {riak_kv_reader, start_link, []},
                 permanent, 30000, worker, [riak_kv_reader]},
 
+    TokenManager = {riak_kv_token_manager,
+                    {riak_kv_token_manager, start_link, []},
+                    permanent, 3000, worker, [riak_kv_token_manager]},
```

Comment:
A short shutdown timeout here...
This means that when you shutdown a node, all ongoing token_sessions will be released, but ongoing work may possibly continue a bit more, since other processes are slighty slower to stop working.
This seems like the right approach to me.